beamer/agent/chain.py

```diff
@@ -636,6 +636,11 @@ def on_future_done(f: Future) -> None:
             context.logger.error("Optimism prove failed", ex=ex, tx_hash=proof_tx)
         else:
             claim.message_proved = True
+            assert request is not None
+            if request.fill_timestamp is not None:
+                request.fill_timestamp = Timestamp(int(time.time()))
+            elif claim.invalidation_timestamp is not None:
+                claim.invalidation_timestamp = Timestamp(int(time.time()))
```

Comment:
I would prefer if we went further than this.
For instance, we have
```
def _proof_ready_for_l1_relay(request: Request) -> bool:
    return (                                            
        not request.l1_resolved.is_active               
        and request.fill_tx is not None                 
        and request.fill_timestamp is not None          
    )                                                   
```
Checking for "fill timestamp" does not make sense in the context of proof readiness anymore.
I would consider replacing `request.fill_timestamp` with something like `request.proof_timestamp`, which would be `None` until the proof is actually done.

Reply:
Additionally, there is

```python
if request.fill_timestamp:                                      
    timestamp_finalized = _timestamp_is_l1_finalized(           
        request.fill_timestamp, context, request.target_chain_id
    )                                                           
    assert request.fill_tx is not None                          
    proof_tx = request.fill_tx                                  
```
in `maybe_resolve`. Note how we set `proof_tx` to be `request.fill_tx`. The question here is, are we still okay with naming like this? I _think_ the code should still continue to work, but it may be misleading which "proof" we are talking about.

Reply:
will change it to resolution_tx

Reply:
> will change it to resolution_tx

I'm not sure that's much better because that could also mean the tx on L1.
What we're talking about is really the original tx on L2 (on the transfer's target chain).

Reply:
resolvable_tx :D